rasa/core/test.py

```diff
@@ -76,32 +76,25 @@ def has_prediction_target_mismatch(self):
             or self.action_predictions != self.action_targets
         )
 
-    def serialise_targets(
-        self, include_actions=True, include_intents=True, include_entities=False
-    ):
-        targets = []
-        if include_actions:
-            targets += self.action_targets
-        if include_intents:
-            targets += self.intent_targets
-        if include_entities:
-            targets += self.entity_targets
-
-        return [json.dumps(t) if isinstance(t, dict) else t for t in targets]
-
-    def serialise_predictions(
-        self, include_actions=True, include_intents=True, include_entities=False
-    ):
-        predictions = []
+    def serialise(self):
+        """Turn targets and predictions to lists of equal size for sklearn"""
+
+        targets = (
+            self.action_targets
+            + self.intent_targets
+            + [json.dumps(t) for t in self.entity_targets]
+        )
+        predictions = (
+            self.action_predictions
+            + self.intent_predictions
+            + [json.dumps(p) for p in self.entity_predictions]
+        )
 
-        if include_actions:
-            predictions += self.action_predictions
-        if include_intents:
-            predictions += self.intent_predictions
-        if include_entities:
-            predictions += self.entity_predictions
+        # sklearn does not cope with lists of unequal size, nor None values
+        padding = len(targets) - len(predictions)
```

Comment:
As I'm not very familiar with this part of the code: Are we sure that targets is always equal or larger then predictions?

Reply:
Good catch